core/src/toga/widgets/button.py

```diff
@@ -37,19 +37,26 @@ def __init__(
 
     @property
     def text(self):
-        """The text displayed on the button."""
-        return self._impl.get_text()
+        """The text displayed on the button.
+
+        ``None`` will be displayed as an empty string. Any other object will be converted
+        to a string using ``str()``.
+
+        Only one line of text can be displayed. Any content after the first newline will
+        be ignored.
+        """
+        return self._text
 
     @text.setter
     def text(self, value):
-        if not value:
-            raise ValueError("Button must have a label")
-
-        # Button text can't include line breaks. Strip any content
-        # after a line break (if provided)
-        value = str(value).split("\n")[0]
+        if value is None:
+            self._text = ""
+        else:
+            # Button text can't include line breaks. Strip any content
+            # after a line break (if provided)
+            self._text = str(value).split("\n")[0]
 
-        self._impl.set_text(value)
+        self._impl.set_text(self._text)
```

Comment:
I'm generally against the introduction of interface state/shadow variables; in this case, the shadow variable is only required if we have problems round-tripping the original value. In this implementation we're not round tripping anyway (because we're losing `None` and multiline input); to support the winforms edge case, I'm entirely comfortable with a normalizing a user-supplied `"\u0200"` (zero width space) to `""`). It's highly unlikely to ever be user supplied in that form, and it's functionally indistinguishable from "".

Reply:
> I'm entirely comfortable with a normalizing a user-supplied `"\u0200"` (zero width space) to `""`).

Makes sense, but \u0200 is LATIN CAPITAL LETTER A WITH DOUBLE GRAVE. I've replaced that with \u200B.